@xen-orchestra/backups/_cleanVm.js

```diff
@@ -528,6 +528,12 @@ exports.cleanVm = async function cleanVm(
     }
   })
 
+  const vmUuid = basename(vmDir)
+  // purge cache
+  await this.invalidateVmBackupListCache(vmUuid)
+  // pregenerate cache
+  await this.listVmBackups(vmUuid)
```

Comment:
I think this is a very common case, worth optimizing.

What do you think of directly inserting the new backup in the existing VM and global caches if they exists?

And still keeping the invalidation on backup deletion and lazy generation on listing.

I think it would be worth it but only if easy to do and without much code duplication.